### DISCUSSION TITLE
How to do scoped registrations?

### DISCUSSION THREAD

**Discussion by jhimes144** [reactiveui/splat#1070] [Q&A]
Hi, could someone point me in the direction on how to work with scoped registrations? I see how to register transient and singletons... but cannot find any details on how to register scoped services.

**Comment by ChrisPulman** [comment]
https://www.reactiveui.net/docs/handbook/dependency-inversion/
Hopefully this will help, if you need more information please shout.

**Comment by Snailya** [comment]
I use the [Splat.Microsoft.Extensions.DependencyInjection](https://www.nuget.org/packages/Splat.Microsoft.Extensions.DependencyInjection/) to help create scoped registrations.
Then used the global Continaer property to create scope.
`public static IServiceProvider Container { get; private set; }`
in other class
`using var scope = Container .CreateScope(){}`

However, this appoach has a bug if you resolve the service using Locater.Current.GetService for an IDisposable service. It can't detect it is IDisposable so it would not call Dispose method after the scope is disposed.
